src/main/kotlin/blackjack/domain/CardPack.kt

```diff
@@ -0,0 +1,32 @@
+package blackjack.domain
+
+import blackjack.enums.CardShape
+import blackjack.enums.CardType
+
+class CardPack {
+    private val cards = mutableListOf<Card>()
+
+    init {
+        CardShape.values().forEach { cardShape ->
+            createCardsOf(cardShape)
+        }
+        cards.shuffle()
+    }
+
+    fun pickCard(): Card {
+
+        if (cards.isEmpty()) {
+            throw IllegalStateException("모든 카드가 사용되었습니다.")
+        }
+
+        val pickedCard = cards[0]
+        cards.remove(pickedCard)
+        return pickedCard
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
        return cards.removeAt(0)
```